scripts/test-projects/src/main.rs

```diff
@@ -0,0 +1,109 @@
+//! Build or format all projects under the main test suite with `forc`/`forc-fmt`
+//!
+//! NOTE: This expects `forc`, `forc-fmt` and `cargo` to be available in `PATH`.
+
+use clap::Parser;
+use futures_util::StreamExt;
+use std::path::PathBuf;
+use test_projects::{run_recursively, BuildResult, Cli, Command2Run, Commands, ResultWriter};
+
+const TESTS_PATH: &str = "packages/fuels/tests/";
+
+#[tokio::main]
+async fn main() {
+    let cli = Cli::parse();
+
+    let mut result_writer = ResultWriter::new();
+
+    let project_path = cli
```

Comment:
`main` seems to be doing a bit too much inline. 

I'd add a step converting raw CLI input into one ready for consumption. You can move all the path `canonicalize` s, binary path resolving, command invocation preparation, etc. Let main have it only after you're sure it's valid.

also, CLI output can be moved into a `print_success` or `print_failure` so that `main` offers the birds-eye-view of what is happening and, if you're interested, you can go and dig deeper into any part you want.

I see three steps here:
1. determine what the CLI user wanted
2. do it
3. report on final success/failure.

so ideally that many lines in `main` :)